lib/puppet/defaults.rb

```diff
@@ -129,6 +129,11 @@ def self.default_vendormoduledir
       :hook => proc {|value| Puppet::Util::Log.level = value },
       :call_hook => :on_initialize_and_write,
     },
+    :coalesce_class_failures => {
```

Comment:
we have a few other "warning" related puppet settings, so perhaps something like `warn_once_class_failures`?

Reply:
Yes but how about `coalesce_class_warnings`?

Reply:
(trying to keep the name "near" `disable_warnings` but if you a strong opinion ...

Reply:
... I'm not sure I could spell coalesce if my life depended on it. How about `merge_class_warnings` or some other synonym? though since this is really about dependencies, maybe `merge_dependency_warnings`?

Reply:
I'll admit to using a spell-checker :) 
Okay .. how about ... `merge_class_dependency_warnings` ?